### ISSUE TITLE
Add support for callbacks in TorchForecastingModel

### ISSUE THREAD

**Issue by alexkyllo** [unit8co/darts#577]
**Is your feature request related to a current problem? Please describe.**

Currently there is no way to inject user code into the training loop (`_train()` method) of `TorchForecastingModel` and its subclasses; this is a major limitation. Callbacks are useful, and in some cases critical, because they allow the developer to do things like pruning the hyperparameter search space via early stopping (halting training if the validation loss has not improved in the last `n` epochs). This is needed to use the models with a hyperparameter optimization framework such as Optuna.

**Describe proposed solution**

I suggest enhancing `TorchForecastingModel` to accept a list of callbacks that will be called at the end of each training epoch. One possibility would be incorporating PyTorch Lightning integration into Darts, to take advantage of its existing callback system. Related documentation: https://pytorch-lightning.readthedocs.io/en/latest/extensions/callbacks.html

**Comment by hrzn** [comment]
Yes, this is definitely important and on the roadmap. It could indeed be a good idea to rely on PyTorch Lightning here.
We plan to tackle this in the future, but here too, we are also welcoming contributions :)

**Comment by hrzn** [comment]
@alexkyllo v0.17.0 released today is based on Pytorch Lightning and supports all its callback methods.

**Comment by alexkyllo** [comment]
> @alexkyllo v0.17.0 released today is based on Pytorch Lightning and supports all its callback methods.

Fantastic--this feature dramatically increases the chances that I would use Darts for my next forecasting project and recommend it to others. Thanks for letting me know!
